.github/workflows/build.yml

```diff
@@ -93,9 +93,19 @@ jobs:
       python-version: ${{ steps.matrix.outputs.python-version }}
       numpy-version: ${{ steps.matrix.outputs.numpy-version }}
     steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
```

Comment:
fetch-depth: 0 fetches "all history for all tags and branches" ref: readme https://github.com/actions/checkout

My understanding is that this is only used to get the last commit message so is unnecessary to pull all of this.

Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. To me this sounds like enough so I would suggest testing without this fetch-depth
```suggestion

```